### PR TITLE
731 refactor tests for tidy methods

### LINKED ISSUE #731: refactor tests for `tidy` methods
It appears that `expect_snapshot` effectively compares the output of `print(<some object)>` to assess test success or failure.  This the result is dependent on both operating system and subtle changes in other packages (such as `pillar`, `color` and `vctrs` when comparing `tibble`s).  This appears to be causing recently written tests on `tidy` methods to fail when comparing snapshots pulled from GitHub with results of the same test run locally.

Therefore, I suggest changing from `expect_snapshot` to `expect_snapshot_value` when testing `tidy` methods.

Quoting from the online doc: "`expect_snapshot()` runs code as if you had executed it at the console, and records the results, including output, messages, warnings, and errors. If you just want to compare the result, try `expect_snapshot_value()`".

### PR DESCRIPTION
**Pull request by Puzzled-Face**
closes #731

### CODE REVIEW

**Review by danielinteractive — COMMENTED** [review]
Thanks @Puzzled-Face , please see below - in general if we can try to separate unrelated updates in separate PRs that would be great, if it is easier to keep in here then ok for this one

File: tests/testthat/test-Rules-methods.R [hunk 1]
```diff
@@ -44,29 +44,7 @@ test_that("nextBest-NextBestMTD returns correct next dose and plot (no doselimit
   vdiffr::expect_doppelganger("Plot of nextBest-NextBestMTD without doselimit", result$plot)
 })
 
-test_that("nextBest-NextBestMTD returns correct next dose and plot when doselimit=0", {
```
[the remark this answers is not in the document]
Comment (Puzzled-Face) [reply tests/testthat/test-Rules-methods.R:47]: Oops!  Can you show me an easy way to recover this, please?

File: examples/mcmc-DataOrdinal.R [hunk 2]
```diff
@@ -1,11 +0,0 @@
-data <- DataOrdinal(
```
Comment (danielinteractive) [inline examples/mcmc-DataOrdinal.R:1]: is it correct that this file is deleted?

File: tests/testthat/test-Rules-methods.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (danielinteractive) [reply tests/testthat/test-Rules-methods.R:47]: easiest will be to just copy / paste it from the main branch, e.g. browsing to https://github.com/Roche/crmPack/blob/main/tests/testthat/test-Rules-methods.R#L47

[the remark this answers is not in the document]
Comment (Puzzled-Face) [reply tests/testthat/test-Rules-methods.R]: Doh!  I figured that out as soon as I pressed send...

File: examples/mcmc-DataOrdinal.R (hunk 2 above)
Comment (Puzzled-Face) [reply examples/mcmc-DataOrdinal.R:1]: Yes.  I will restore it in #707.

**Review by danielinteractive — APPROVED** [review]

**Pull request merged**